src/inmanta/util/dict_path.py

```diff
@@ -489,6 +538,139 @@ def get_elements(self, container: object) -> list[object]:
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                KeyedList("relation", "key_attribute", "key_value"),
+                KeyedList("relation", "key_attribute", "other_value"),
+            ]
+
+            assert WildKeyedList("relation",[("key_attribute","*"), ("other_attribute", "*")]).get_paths(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                KeyedList("relation", [("key_attribute","key_value"), ("other_attribute", "other_value")]),
+            ]
+
+            assert WildKeyedList("relation",[("*","other_value")]).get_paths(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                KeyedList("relation", [("other_attribute","other_value")]),
+                KeyedList("relation", [("key_attribute","other_value")]),
+                KeyedList("relation", [("other_key_attribute","other_value")]),
+            ]
+
+            assert WildKeyedList("relation",[("*","*")]).get_paths(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                KeyedList("relation", [("key_attribute","key_value")]),
```

Comment:
I don't see how this works with the described usecase? 

I expect, from the explanation, that this would be used to find a difference. 

let say we go from the set you have now to one that adds 
```python
"relation":[
                        {
                            "key_attribute":"key_value",
                            "other_attribute":"other_value",
                        },
                        {
                            "key_attribute":"other_value",
                        },
                        {
                            "other_key_attribute":"other_value",
                        },
                        {
                           "key_attribute":"other_value",
                           "other_key_attribute":"other_value",
                        },
                    [
                        
```

We try to detect change by doing 

```python
  removed_entitites = (
        {str(p) for p in parsed_wild_path.get_paths(previous_attributes)}
        - {str(p) for p in parsed_wild_path.get_paths(current_attributes)}
    )
```
Which is empty 


i.e. this filter is much weaker then expected, in that it doesn't identify any individual members. 

I think this is because the interface definition is to weak/wide for the intended use case.

Reply:
> I don't see how this works with the described usecase?

It doesn't, I couldn't think of a use case where we would want to use the wild card on the key.  But I didn't feel like ignoring this usage of the wild keyed list either.  So I added support for it and illustrated it in the example.

To refer to your other comment:
> I think you have to assume the path describes an identity and that identity is made concrete based on the instances present.

I completely agree with that, in the use case I have it is the only thing that makes sense, but it seemed easier to me to implement the full, and less strict functionality, leaving the responsibility to the developer of making a good usage of it.